src/server/semester/semester.service.ts

```diff
@@ -59,4 +62,150 @@ export class SemesterService {
+        .map((fallInstance) => {
+          const springInstance: CourseInstance = courseIdToSpringInstance[
+            fallInstance.course.id
+          ];
+          return {
+            ...new CourseInstance(),
+            offered: springInstance.offered === OFFERED.RETIRED
+              ? OFFERED.RETIRED : OFFERED.BLANK,
+            preEnrollment: null,
+            studyCardEnrollment: null,
+            actualEnrollment: null,
+            course: fallInstance.course,
+            facultyCourseInstances: [],
+            meetings: [],
+          };
+        });
+
+      // Create new fall semester
+      const fallSemester = new Semester();
+      // NOTE: academic year in Semester entity is actually the calendar year
+      fallSemester.academicYear = (newAcademicYear - 1).toString();
+      fallSemester.term = TERM.FALL;
+      fallSemester.courseInstances = newFallInstances;
+
+      fallSemester.nonClassEvents = existingFallSemester.nonClassEvents
+        .map((nce) => ({
+          ...new NonClassEvent(),
+          nonClassParent: nce.nonClassParent,
+          private: nce.private,
+          meetings: [],
+        }));
+
+      fallSemester.absences = existingFallSemester.absences
+        .map((absence) => ({
+          ...new Absence(),
+          faculty: absence.faculty,
+          type: ABSENCE_TYPE.PRESENT,
+        }));
+
+      // Note that this also saves the nested entities because of
+      // the cascade set on semester
+      await this.semesterRepository.save(fallSemester);
+
+      const newSpringInstances = springInstances
+        .map((springInstance) => ({
+          ...new CourseInstance(),
+          offered: springInstance.offered === OFFERED.RETIRED
+            ? OFFERED.RETIRED : OFFERED.BLANK,
+          preEnrollment: null,
+          studyCardEnrollment: null,
+          actualEnrollment: null,
+          course: springInstance.course,
+          facultyCourseInstances: [],
+          meetings: [],
+        }));
+
+      // Create new spring semester
+      const springSemester = new Semester();
+      springSemester.academicYear = newAcademicYear.toString();
+      springSemester.term = TERM.SPRING;
+      springSemester.courseInstances = newSpringInstances;
+
+      springSemester.nonClassEvents = existingSpringSemester.nonClassEvents
+        .map((nce) => ({
+          ...new NonClassEvent(),
+          nonClassParent: nce.nonClassParent,
+          private: nce.private,
+          meetings: [],
+        }));
+
+      springSemester.absences = existingSpringSemester.absences
+        .map((absence) => ({
+          ...new Absence(),
+          faculty: absence.faculty,
+          type: ABSENCE_TYPE.PRESENT,
+        }));
+
+      await this.semesterRepository.save(springSemester);
+    }
+    return null;
+  }
+
+  public async onApplicationBootstrap(): Promise<void> {
+    const today = new Date();
+    if (today.getMonth() === 5) {
+      await this.addAcademicYear(today.getFullYear() + 1);
```

Comment:
That makes sense. This was fixed so that the `await` was removed in [b5af716](https://github.com/seas-computing/course-planner/pull/496/commits/b5af716bfd7c2b5713b977d5a6b9795e6994ff2a).